### ISSUE TITLE
Avoid exposing client keystore and truststore passwords in the JDBC URL

### ISSUE THREAD

**Issue by hamdiazz** [TOSIT-IO/tdp-collection#832]
Hello,
At present, we have in beeline-site.xml:
```
beeline.hs2.jdbc.url.zk_cluster: jdbc:hive2://{{ hive_site["hive.zookeeper.quorum"] }}/;serviceDiscoveryMode=zooKeeper;zooKeeperNamespace=hiveserver2;principal={{ hiveserver2_site["hive.server2.authentication.kerberos.principal"] }};sslTrustStore={{ hive_truststore_location }};trustStorePassword={{ hive_truststore_password }}
and as you can see, we have hive_truststore_password in the JDBC URL. Exposing this password in URL can be a security concern. We can hide this password from JDBC URL when we protect these password in a local JCEKS keystore file and pass the JCEKS file path to URL instead.
```

As below, i tested this JDBC URL and it works:
`beeline -u 'jdbc:hive2://{{ hive_site["hive.zookeeper.quorum"] }}/;serviceDiscoveryMode=zooKeeper;zooKeeperNamespace=hiveserver2;principal={{ hiveserver2_site["hive.server2.authentication.kerberos.principal"] }};ssl=true;sslKeyStore=/etc/ssl/certs/keystore.jks;storePasswordPath=localjceks://file/etc/hive/conf/hive.jceks'`
What do you think ?

Thanks,

**Comment by rpignolet** [comment]
The truststore only contain public certificates and to modify it you must have the write permission on the truststore file so it is not a security issue to expose the password of the truststore (see #98).

You replace the truststore by the keystore in the URL and public certificate should not be inside the keystore so I don't understand why your proposition works.

**Comment by hamdiazz** [comment]
Thanks, i found this Jira Hive also https://issues.apache.org/jira/browse/HIVE-27308

**Comment by rpignolet** [comment]
Yes I agree to use `jceks` instead of writing the password in the URL but I don't understand why this is working with a keystore instead of a truststore and to read the key inside the `jceks` we do not need a password too ?

**Comment by Pierrotws** [comment]
Hi @hamdiazz,

This patch is not included in TDP versions of hive, so what do you mean by "deployed it on two dev environment".
Have you deployed apache version of hive or patched TDP ?

If not, the parameter would probably be ignored by beeline. You might check if beeline is working without property. (You may have certificate in JRE default trustore).
In that case, this PR will fail on other setups than yours.

**Comment by hamdiazz** [comment]
Hi @Pierrotws,
I was inspired by the APACHE HIVE PR, but it was deployed on patched TDP.

**Comment by hamdiazz** [comment]
Beeline is working without property.

**Comment by rpignolet** [comment]
The problem being to remain compatible with the existing one, we cannot modify the Ansible code based on the principle that the user has a patch version of Hive.
If there is a release of Hive with this modification, we can make a condition on the deployed version but this is not the case for the moment.

**Comment by rpignolet** [comment]
To be discussed for Hive 4. Closing for now.
